collated/test/A0139257X.md

```diff
@@ -0,0 +1,718 @@
+                String.format(AddCommand.MESSAGE_SUCCESS, toBeAdded),
+                expectedAB,
+                expectedAB.getTaskList());
+    }
+    
+    @Test
+    public void execute_add_ValidTaskWithValidTimeButNoPreFix() throws Exception {
+     // setup expectations
+        TestDataHelper helper = new TestDataHelper();
+        Task toBeAdded = helper.generateFloatingTask("go shopping 7pm", "0");
+        TaskManager expectedAB = new TaskManager();
+        expectedAB.addTask(toBeAdded);
+
+        // execute command and verify result
+        assertCommandBehavior(helper.generateAddFloatingTaskCommand(toBeAdded),
+                String.format(AddCommand.MESSAGE_SUCCESS, toBeAdded),
+                expectedAB,
+                expectedAB.getTaskList());
+    }
+    
+    @Test
+    public void execute_add_ValidTaskWithImproperUsageOfOn() throws Exception {
+     // setup expectations
+        TestDataHelper helper = new TestDataHelper();
+        Task toBeAdded = helper.generateFloatingTask("go shopping on 7pm", "0");
+        TaskManager expectedAB = new TaskManager();
+        expectedAB.addTask(toBeAdded);
+
+        // execute command and verify result
+        assertCommandBehavior(helper.generateAddFloatingTaskCommand(toBeAdded),
+                String.format(AddCommand.MESSAGE_SUCCESS, toBeAdded),
+                expectedAB,
+                expectedAB.getTaskList());
+    }
+    
+    @Test
+    public void execute_add_ValidTaskWithMultipleOn() throws Exception {
+        String description = "add go shopping on 2-2-2222 on 3-3-3333";
+        Task toBeAdded = new EventTask("go shopping on 3-3-3333", "2-2-2222", "2-2-2222", TaskTime.DEFAULT_START_TIME, TaskTime.DEFAULT_END_TIME, "0", TaskStatus.INCOMPLETE, new UniqueTagList());
+        TaskManager expectedAB = new TaskManager();
+        expectedAB.addTask(toBeAdded);
+        // execute command and verify result
+        assertCommandBehavior(description,
+                String.format(AddCommand.MESSAGE_SUCCESS, toBeAdded),
+                expectedAB,
+                expectedAB.getTaskList());
+    }
+    
+    @Test
+    public void execute_add_ValidTaskWithMultipleAt() throws Exception {
+        String description = "add go shopping at 11.58pm at 11.59pm";
+        Task toBeAdded = new EventTask("go shopping at 11.59pm", TaskDate.DEFAULT_DATE, TaskDate.DEFAULT_DATE, "11.58pm", TaskTime.DEFAULT_END_TIME, "0", TaskStatus.INCOMPLETE, new UniqueTagList());
+        TaskManager expectedAB = new TaskManager();
+        expectedAB.addTask(toBeAdded);
+        // execute command and verify result
+        assertCommandBehavior(description,
+                String.format(AddCommand.MESSAGE_SUCCESS, toBeAdded),
+                expectedAB,
+                expectedAB.getTaskList());
+    }
+    
+    @Test
+    public void execute_add_ValidTaskWithMultipleByDate() throws Exception {
+        String description = "add go shopping by 2-2-2222 by 3-3-3333";
```

Comment:
avoid magic strings like this; please take a look at how addressbook uses a command builder for tests